circuits/src/transaction.rs

```diff
@@ -408,9 +263,44 @@ impl<const H: usize, const I: usize> TxCircuit<H, I> {
             sender_blinder,
         }
     }
+
+    /// Return the public-input values used for the circuit
+    pub fn public_inputs(&self) -> Vec<BlsScalar> {
```

Comment:
why is this needed if that's returned by the prover?

Reply:
because in practice the verifier doesn't have access to the data that the prover returns.

Reply:
This got me thinking, I'll remove the function again!